<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="syntenyPlotteR">
<title>SyntenyPlotteR - a tool for genome synteny visualisation  • syntenyPlotteR</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="SyntenyPlotteR - a tool for genome synteny visualisation ">
<meta property="og:description" content="syntenyPlotteR">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">syntenyPlotteR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.0.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item">
  <a class="nav-link" href="../articles/syntenyPlotteR.html">Get started</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/Farre-lab/syntenyPlotteR/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>SyntenyPlotteR - a tool for genome synteny visualisation </h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/Farre-lab/syntenyPlotteR/blob/HEAD/vignettes/syntenyPlotteR.Rmd" class="external-link"><code>vignettes/syntenyPlotteR.Rmd</code></a></small>
      <div class="d-none name"><code>syntenyPlotteR.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>Determining syntenic relationships between genomes is a important
step in understanding genome evolution. To do this, researchers align
species genomes to one another to detect sequence markers and conserved
sequences to identify syntenic regions. There is a plethora of tools
which can be used to identify this synteny such as DESCHRAMBLER.
However, they often return tables showing the orientation and, start and
stop positions of the syntenic regions which can often be hard to
interpret at face value.</p>
<p>Here, we provide a tool which visualises this data, improving the
interpretation of the alignment data, particularly for the
identification, characterization, and visualization of chromosome
rearrangements. SyntenyPlotteR relies on the use of ggplot2 to create
the graphics, producing three potential visualizations: a linear
relationship between syntenic regions of the genome using
<code><a href="../reference/draw.linear.html">draw.linear()</a></code>, an evolution highway style grid showing
synteny between single chromosomes aligned between a single reference
species and one or more target species using <code><a href="../reference/draw.eh.html">draw.eh()</a></code>, and
an ideogram with painted chromosomes to show syntenic relationships
between two genomes using <code><a href="../reference/draw.ideogram.html">draw.ideogram()</a></code>. There is also a
function which can reformat data structured in the
DESCHRAMBLER/inferCARs format to a structure suitable for the required
input to the aforementioned functions using
<code><a href="../reference/reformat.syntenyData.html">reformat.syntenyData()</a></code>.</p>
<p>SyntenyPlotteR is solely designed to plot synteny data already
gathered by the user. This tool does not generate syntenic relationships
or curate data. As such, the user is required to construct syntenic
blocks, therefore, allowing the user to choose the most appropriate tool
for their needs. SyntenyPlotteR is easily implemented by the user with
single line functions taking directly the tab delimited files containing
the alignment data and chromosome lengths, thus meaning even
inexperienced users find the process fast to learn and simple to
use.</p>
<p>This package has been developed by Sarah Quigley (<a href="mailto:slq4@kent.ac.uk" class="email">slq4@kent.ac.uk</a>), Joana
Damas (<a href="mailto:joanadamas@gmail.com" class="email">joanadamas@gmail.com</a>), Denis Larkin (<a href="mailto:dlarkin@rvc.ac.uk" class="email">dlarkin@rvc.ac.uk</a>) and
Marta Farré (<a href="mailto:m.farre-belmonte@kent.ac.uk" class="email">m.farre-belmonte@kent.ac.uk</a>)</p>
</div>
<div class="section level2">
<h2 id="to-install">To install:<a class="anchor" aria-label="anchor" href="#to-install"></a>
</h2>
<div class="section level3">
<h3 id="using-cran">Using CRAN<a class="anchor" aria-label="anchor" href="#using-cran"></a>
</h3>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"syntenyPlotteR"</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://farre-lab.github.io/syntenyPlotteR/">syntenyPlotteR</a></span><span class="op">)</span></span></code></pre></div>
<hr>
</div>
</div>
<div class="section level2">
<h2 id="input-files">Input files<a class="anchor" aria-label="anchor" href="#input-files"></a>
</h2>
<div class="section level3">
<h3 id="alignment-output-file-format-for-reformat-syntenydata-function">Alignment output file format for <code>reformat.syntenyData()</code>
function<a class="anchor" aria-label="anchor" href="#alignment-output-file-format-for-reformat-syntenydata-function"></a>
</h3>
<p>If chromosome alignment was performed using DESCHRAMBLER please input
the .map file If chromosome alignment was performed using inferCARs
please input the inferCARs output file</p>
</div>
<div class="section level3">
<h3 id="example-file-format">Example file format<a class="anchor" aria-label="anchor" href="#example-file-format"></a>
</h3>
<div class="figure">
<img src="images/example.deshrambler.output.png" width="380" alt=""><p class="caption">Figure 1 - Example output format from
DESCHRAMBLER/inferCARs</p>
</div>
</div>
<div class="section level3">
<h3 id="alignment-file-input-format-for-draw-eh-draw-ideogram-and-draw-linear-functions">Alignment file input format for <code>draw.eh()</code>,
<code>draw.ideogram()</code>, and <code>draw.linear()</code>
functions<a class="anchor" aria-label="anchor" href="#alignment-file-input-format-for-draw-eh-draw-ideogram-and-draw-linear-functions"></a>
</h3>
<p>Please provide a file for the alignment synteny blocks following this
format, separated by tabs</p>
<p>DO NOT include a header line</p>
<ul>
<li>Reference chromosome ID</li>
<li>Reference start position</li>
<li>Reference end position</li>
<li>Target chromosome</li>
<li>Target start position</li>
<li>Target end position</li>
<li>Orientation</li>
<li>Reference species ID</li>
<li>Target species ID</li>
</ul>
</div>
<div class="section level3">
<h3 id="example-file-format-1">Example file format<a class="anchor" aria-label="anchor" href="#example-file-format-1"></a>
</h3>
<div class="figure">
<img src="images/example.alignment.input.png" width="536" alt=""><p class="caption">Figure 2 - Example input format for alignment data
files showing the alignments of different chromosomes between the
reference sps2 and target sps1.</p>
</div>
<div class="figure">
<img src="images/example.eh.input.png" width="525" alt=""><p class="caption">Figure 3 - Example input format for draw.eh()
function. This figure shows how there are alignments between the
reference species (sps4 - column 8) chromosome 17 (column 1) and
multiple target species (column 9)</p>
</div>
</div>
<div class="section level3">
<h3 id="chromosome-length-file-for-draw-ideogram-and-draw-linear-functions">Chromosome Length file for <code>draw.ideogram()</code> and
<code>draw.linear()</code> functions<a class="anchor" aria-label="anchor" href="#chromosome-length-file-for-draw-ideogram-and-draw-linear-functions"></a>
</h3>
<p>Please provide a file containing all aligned species in order from
first target species in the alignment files followed by the first
reference species in the alignment files – top of file – to the last
target species and reference species in the alignment files – end of
file, following this format, separated by tabs.</p>
<p><em>Note: the order you input the species in the length file
determines the order of which the species will be placed in the
<code><a href="../reference/draw.linear.html">draw.linear()</a></code> function; for example the first species in
the length file (top of file) will be the first linear genome drawn thus
at the bottom of the Y axis. In the case of aligning two target species
to a single reference species, it is suggested you place the reference
in the middle of the length file - i.e. target 1 (top of file),
reference (middle of file), target 2 (bottom of file).</em></p>
<p>DO NOT include a header line</p>
<ul>
<li>Chromosome ID</li>
<li>Chromosome length</li>
<li>Species ID</li>
</ul>
</div>
<div class="section level3">
<h3 id="example-file-format-2">Example file format<a class="anchor" aria-label="anchor" href="#example-file-format-2"></a>
</h3>
<div class="figure">
<img src="images/example.lengths.input.png" width="306" alt=""><p class="caption">Figure 4 - Example input file format for length file
showing the first column containing the chromosome name/number, the
second column containing the chromosome length, and the third column
containing the species name.</p>
</div>
<hr>
</div>
</div>
<div class="section level2">
<h2 id="reformatting-alignment-data">Reformatting alignment data<a class="anchor" aria-label="anchor" href="#reformatting-alignment-data"></a>
</h2>
<p>The syntenyPlotteR package includes a function to reformat alignment
synteny data such as from DESCHRAMBLER or inferCARs - this does not
curate files only reformats it</p>
<p>The function outputs a text file containing the reformatted alignment
data. As default the function saves the file to a temporary directory,
this can be overridden using the <code>directory</code> parameter
i.e. <code>directory = "path/to/directory"</code>.</p>
<div class="section level3">
<h3 id="usage">
<strong>Usage</strong><a class="anchor" aria-label="anchor" href="#usage"></a>
</h3>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://farre-lab.github.io/syntenyPlotteR/">syntenyPlotteR</a></span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="../reference/reformat.syntenyData.html">reformat.syntenyData</a></span><span class="op">(</span><span class="st">"file_data"</span>, <span class="st">"output.filename"</span>, reference.species <span class="op">=</span> <span class="st">"reference.sps"</span>, target.species <span class="op">=</span> <span class="st">"target.sps"</span><span class="op">)</span></span></code></pre></div>
<ul>
<li>file_data - alignment output file from DESCHRAMBLER/infeCARs</li>
<li>output.filename - output file name for reformatted data in a
character string</li>
</ul>
<p>There are optional parameters for some customization of this
function:</p>
<ul>
<li>reference.species - reference species ID as character string for
final output table</li>
<li>target.species - target species ID as character string for final
output table</li>
<li>directory - directory where to save text file
i.e. <code>directory = "path/to/directory"</code>
</li>
</ul>
</div>
<div class="section level3">
<h3 id="example-code-using-data-files-in-instextdata">Example code using data files in <em>inst/extdata/</em><a class="anchor" aria-label="anchor" href="#example-code-using-data-files-in-instextdata"></a>
</h3>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/reformat.syntenyData.html">reformat.syntenyData</a></span><span class="op">(</span><span class="st">"example_map_1.map"</span>, <span class="st">"reformatted.data"</span><span class="op">)</span></span></code></pre></div>
<hr>
</div>
</div>
<div class="section level2">
<h2 id="evolution-highway-style">Evolution Highway style<a class="anchor" aria-label="anchor" href="#evolution-highway-style"></a>
</h2>
<p>The function outputs an image file containing the evolution highway
style alignment. As default the function saves the file to a temporary
directory, this can be overridden using the <code>directory</code>
parameter i.e. <code>directory = "path/to/directory"</code>.</p>
<div class="section level3">
<h3 id="usage-1">Usage<a class="anchor" aria-label="anchor" href="#usage-1"></a>
</h3>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://farre-lab.github.io/syntenyPlotteR/">syntenyPlotteR</a></span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="../reference/draw.eh.html">draw.eh</a></span><span class="op">(</span><span class="st">"output"</span>, <span class="va">chrRange</span>, <span class="st">"data_file"</span>, fileformat <span class="op">=</span> <span class="st">"png"</span>, colour <span class="op">=</span> <span class="st">"lightblue"</span>, inverted.colour <span class="op">=</span> <span class="st">"lightpink"</span>, w <span class="op">=</span> <span class="fl">5.5</span>, h <span class="op">=</span> <span class="fl">10</span>, ps <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></span></code></pre></div>
<ul>
<li>output - String assigned to name of output file name</li>
<li>chrRange - The range of chromosomes of the reference species this is
entered as either a single number i.e. 1 or a range of numbers
i.e. 1:22. <em>Note: if you are inputting sex chromosomes or chromosomes
with characters in the name input a concatenated string
i.e. c(1,2,3,“X”)</em>
</li>
<li>data_file - File containing the syntenic blocks of alignments of one
or more target species aligned to a single reference</li>
</ul>
<p>There are optional parameters for some customization of this
function:</p>
<ul>
<li>fileformat - format for saving the image i.e. png or pdf, parameter
use: <code>fileformat = "pdf"</code> (the default value is “png”)</li>
<li>colour - colour of the syntenic blocks (not inverted blocks),
parameter use: <code>colour = "red"</code> (the default value is
“lightblue”, see Rcolour pallette for colour options)</li>
<li>inverted.colour - colour of the inverted syntenic blocks, parameter
use: <code>inverted.colour = "blue"</code> (the default value is
“lightpink”, see Rcolour pallette for colour options)</li>
<li>w - The width of the image created can be changed by using:
<code>w = 5.5</code> (default)</li>
<li>h - The height of the image created can be changed by using:
<code>h = 10</code> (default)</li>
<li>ps - The point size of the image created can be changed by using:
<code>ps = 10</code> (default)</li>
<li>directory - directory where to save image file
i.e. <code>directory = "path/to/directory"</code> (default is save to
temporary directory)</li>
</ul>
</div>
<div class="section level3">
<h3 id="example-code-using-data-files-in-instextdata-1">Example code using data files in <em>inst/extdata/</em><a class="anchor" aria-label="anchor" href="#example-code-using-data-files-in-instextdata-1"></a>
</h3>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/draw.eh.html">draw.eh</a></span><span class="op">(</span><span class="st">"outputName"</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">17</span>,<span class="st">"X"</span><span class="op">)</span>, <span class="st">"example_eh_alignments_2.txt"</span>, fileformat <span class="op">=</span> <span class="st">"pdf"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="example-output">Example output:<a class="anchor" aria-label="anchor" href="#example-output"></a>
</h3>
<div class="figure">
<img src="images/example_eh.png" width="258" alt=""><p class="caption">Figure 5 - Example alignment in Evolution Highway
style. This figure shows chromosome 17 from a single reference species
aligned to three target species - sps1, sps2, and sps3. Sps1 and sps2
chromosome 17 shows synteny with the reference chromosome 17 whereas
sps3 chromosome 16 is syntenic to the reference chromosome 17. The plot
represents co-linear synteny with the blue shading and inverted synteny
with the pink shading. White space represents alignment gaps.</p>
</div>
<hr>
</div>
</div>
<div class="section level2">
<h2 id="chromosome-painting-style">Chromosome painting style<a class="anchor" aria-label="anchor" href="#chromosome-painting-style"></a>
</h2>
<p>The function outputs an image file containing the chromosome painting
style alignment. As default the function saves the file to a temporary
directory, this can be overridden using the <code>directory</code>
parameter i.e. <code>directory = "path/to/directory"</code>.</p>
<div class="section level3">
<h3 id="usage-2">Usage<a class="anchor" aria-label="anchor" href="#usage-2"></a>
</h3>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://farre-lab.github.io/syntenyPlotteR/">syntenyPlotteR</a></span><span class="op">)</span>  </span>
<span></span>
<span><span class="fu"><a href="../reference/draw.ideogram.html">draw.ideogram</a></span><span class="op">(</span><span class="st">"file_data"</span>, <span class="st">"sizefile"</span>, <span class="st">"output"</span>, fileformat <span class="op">=</span> <span class="st">"png"</span>, colours <span class="op">=</span> <span class="va">colours.default</span>, w<span class="op">=</span><span class="fl">8.5</span>, h<span class="op">=</span><span class="fl">10</span>, ps<span class="op">=</span><span class="fl">5</span><span class="op">)</span></span></code></pre></div>
<ul>
<li>file_data - file containing the syntenic blocks</li>
<li>sizefile - tab separated file of all chromosome, scaffold, or contig
lengths and the species identifier, in order from first target species
in the alignment files followed by the first reference species in the
alignment files – top of file – to the last target species and reference
species in the alignment files – end of file.</li>
<li>output - string assigned to the output file name</li>
</ul>
<p>There are optional parameters for some customization of this
function:</p>
<ul>
<li>fileformat - format for saving the image i.e. png or pdf, parameter
use: <code>fileformat = "pdf"</code> (the default value is “png”)</li>
<li>colours - colours to assign to the ideograms in a concatenated
string of chromosome IDs with assigned colour values which can be found
with R colour Pallette, paramter use:
<code>colours = c("1" = "red", "2" = "blue", "3" = "green","4" = "orange", "5" = "purple","X" = "grey")</code>
if no colours are assigned default values will be used but colours MUST
be assigned to all chromosomes</li>
<li>w - The width of the image created can be changed by using:
<code>w = 5.5</code> (default)</li>
<li>h - The height of the image created can be changed by using:
<code>h = 10</code> (default)</li>
<li>ps - The point size of the image created can be changed by using:
<code>ps = 10</code> (default)</li>
<li>directory - directory where to save image file
i.e. <code>directory = "path/to/directory"</code> (default is save to
temporary directory)</li>
</ul>
<p>Target is the species which chromosomes will be painted. Reference
will be used for painting and diagonals. Chromosomes will be in the same
order as in the target sizes file.</p>
<p><em>Note: If you have any alignments with chromosomes not defined in
the length file the function will not run.</em></p>
</div>
<div class="section level3">
<h3 id="example-code-using-data-files-in-instextdata-2">Example code using data files in <em>inst/extdata/</em><a class="anchor" aria-label="anchor" href="#example-code-using-data-files-in-instextdata-2"></a>
</h3>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/draw.ideogram.html">draw.ideogram</a></span><span class="op">(</span><span class="st">"example_alignment_1.txt"</span>, <span class="st">"example_lengths.txt"</span>, <span class="st">"example.ideogram"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="example-output-1">Example output:<a class="anchor" aria-label="anchor" href="#example-output-1"></a>
</h3>
<div class="figure">
<img src="images/example.ideogram.png" width="360" alt=""><p class="caption">Figure 6 - Example ideogram showing a comparison
between sps1 and sps2. In this figure sps1 is the target genome used to
draw the chromosome ideograms, and sps2 is the reference genome. Where
there is shared synteny between sps1 and sps2 the colours assigned to
the sps2 chromosomes are used to colour the syntenic regions i.e. sps1
chromosome 1 is drawn using the black box and the green shading shows
synteny with sps2 chromosome 1. Each syntenic block is traversed by a
diagonal line, identification of multiple syntenic blocks is shown where
there is a vertical line representing the end of one block and a start
of another, with a diagonal line in each syntenic block, whereas
completely syntenic chromosomes are depicted with a single line drawn
from the bottom left to top right such as with chromosome 1. Inverted
syntenic blocks are represented with a diagonal drawn from the top left
to the bottom right as seen in the first syntenic block of chromosome 19
- sps1 chromosome 19 ideogram drawn and coloured showing synteny with
sps2 chromosome 18.</p>
</div>
<hr>
</div>
</div>
<div class="section level2">
<h2 id="linear-style">Linear style<a class="anchor" aria-label="anchor" href="#linear-style"></a>
</h2>
<p>The function outputs an image file containing the linear style
alignment. As default the function saves the file to a temporary
directory, this can be overridden using the <code>directory</code>
parameter i.e. <code>directory = "path/to/directory"</code>.</p>
<div class="section level3">
<h3 id="usage-3">Usage<a class="anchor" aria-label="anchor" href="#usage-3"></a>
</h3>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://farre-lab.github.io/syntenyPlotteR/">syntenyPlotteR</a></span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="../reference/draw.linear.html">draw.linear</a></span><span class="op">(</span><span class="va">output</span>, <span class="va">sizefile</span>, <span class="va">...</span>, fileformat <span class="op">=</span> <span class="st">"png"</span>, colours <span class="op">=</span> <span class="va">colours.default</span>, w<span class="op">=</span><span class="fl">13</span>, h<span class="op">=</span><span class="fl">5</span>, opacity <span class="op">=</span> <span class="fl">.5</span><span class="op">)</span></span></code></pre></div>
<ul>
<li>output - string assigned to the output file name</li>
<li>sizefile - tab separated file of all chromosome, scaffold, or contig
lengths and the species identifier, in order from first target species
in the alignment files followed by the first reference species in the
alignment files – top of file – to the last target species and reference
species in the alignment files – end of file.</li>
<li>… - files containing the syntenic blocks (one file per alignment, in
order from first target/reference (most recent species pairwise
alignment in ancestral reconstruction data) alignment file to last
target/reference (ancestor pairwise alignment in ancestral
reconstruction data) alignment file)</li>
</ul>
<p>Please ensure any species identifiers used between length and
alignment files are matching (same identifiers and letter case)</p>
<p>There are optional parameters for some customization of this
function:</p>
<ul>
<li>fileformat - format for saving the image i.e. png or pdf, parameter
use: <code>fileformat = "pdf"</code> (the default value is “png”)</li>
<li>colours - colours to assign to the bands between ideograms in a
concatenated string of chromosome IDs with assigned colour values which
can be found with R colour Pallette, paramter use:
<code>colours = c("1" = "red", "2" = "blue", "3" = "green","4" = "orange", "5" = "purple","X" = "grey")</code>
if no colours are assigned default values will be used but colours MUST
be assigned to all chromosomes</li>
<li>w - The width of the image created can be changed by using:
<code>w = 5.5</code> (default)</li>
<li>h - The height of the image created can be changed by using:
<code>h = 10</code> (default)</li>
<li>opacity - the opacity of the ribbons can be changes using inputting:
<code>opacity = .5</code> (default)</li>
<li>directory - directory where to save image file
i.e. <code>directory = "path/to/directory"</code> (default is save to
temporary directory)</li>
</ul>
<p><em>Note: If you have any alignments with chromosomes not defined in
the length file the function will not run.</em></p>
</div>
<div class="section level3">
<h3 id="example-code-using-data-files-in-instextdata-3">Example code using data files in <em>inst/extdata/</em><a class="anchor" aria-label="anchor" href="#example-code-using-data-files-in-instextdata-3"></a>
</h3>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/draw.linear.html">draw.linear</a></span><span class="op">(</span><span class="st">"example_linear"</span>, <span class="st">"example_lengths.txt"</span>, <span class="st">"example_alignment_1.txt"</span>, <span class="st">"example_alignment_2.txt"</span>, <span class="st">"example_alignment_3.txt"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="example-output-2">Example output:<a class="anchor" aria-label="anchor" href="#example-output-2"></a>
</h3>
<div class="figure">
<img src="images/example_linear.png" width="530" alt=""><p class="caption">Figure 7 - Example linear alignment plot between four
species, from sps4 through to sps1. The grey boxes represent the
chromosomes in each species genome, with the chromosome number written
above the box. The coloured bands represent synteny between the genomes
directly above/below one another. The colours of the bands represent the
reference chromosome that shares synteny with the target chromosome
i.e. chromosome 3 was assigned the colour pink thus any synteny
originating from chromosome 3 in either sps2, sps3, or sps4 to the
genome below will be shown by a pink band.</p>
</div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Marta Farre, Sarah Quigley, Joana Damas, Denis Larkin.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
